File: unit-3/sprint-3/day-1/assignments/Assignment_Playmusic/app/src/main/java/com/example/playmusic_usingservice/MainActivity.java
package com.example.playmusic_usingservice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button mBtnStart, mBtnStop;
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        mBtnStart = findViewById(R.id.btnStart);
        mBtnStop = findViewById(R.id.btnStop);
        intent = new Intent(this, MusicService.class);
        mBtnStop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                stopService(intent);
            }
        });

        mBtnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService(intent);

            }
        });
    }
}

File: unit-3/sprint-4/day-2/you/GetUserDetails/app/src/main/java/com/example/a18augyougetuserdetails/MainActivity.kt
package com.example.a18augyougetuserdetails

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnFetch.setOnClickListener{
            callApi()
        }
    }

    private fun callApi() {
        val apiService = Network.getInstance().create(ApiService::class.java)
        apiService.getUser(etId.text.toString().toInt())
            .enqueue(object : retrofit2.Callback<ResponseModel>{
                override fun onResponse(
                    call: Call<ResponseModel>,
                    response: Response<ResponseModel>
                ) {
                    val model = response.body()
                    tvEmail.text = model!!.data.email
                    tvCompany.text = model?.data.firstName
                }
                override fun onFailure(call: Call<ResponseModel>, t: Throwable) {
                    Toast.makeText(this@MainActivity,"Fail to load data",Toast.LENGTH_SHORT).show()
                }

            })
    }


}